app/presenters/provider/admin/dashboard_presenter.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class Provider::Admin::DashboardPresenter
+  include System::UrlHelpers.system_url_helpers
+
+  WIDGET_SIZE = 5
+
+  def initialize(user:)
+    @user = user
+    @ability = Ability.new(user)
+
+    @products = user.accessible_services
+                    .order('updated_at desc')
+                    .limit(WIDGET_SIZE)
+
+    @backend_apis = user.accessible_backend_apis
+                        .order('updated_at desc')
+                        .limit(WIDGET_SIZE)
+
+    @service_actions_presenter = ServiceActionsPresenter.new(user)
+  end
+
+  attr_reader :products, :backend_apis
+
+  def products_widget_data
+    {
+      products: products.map do |product|
+        {
+          id: product.id,
+          name: product.name,
+          updated_at: product.updated_at.to_s(:long),
+          link: product.decorate.link,
+          links: service_actions_presenter.actions(product)
+        }
+      end,
+      newProductPath: can_create_service ? new_admin_service_path : nil,
+      productsPath: admin_services_path
+    }
+  end
+
+  # TODO: for some reason, accessible_backend_apis is not empty even though can?(:see, BackendApi) is false
```

Comment:
I think this scope is just `BackendApi.where.not(state: DELETED_STATE)`, but I might be wrong...

Reply:
That makes sense